src/main/java/org/betonquest/betonquest/database/MySQL.java

```diff
@@ -143,70 +158,89 @@ private void migration1(final Connection connection) throws SQLException {
-                    + "(SELECT MIN(id) FROM " + prefix + "points GROUP BY playerID, category)");
+            statement.executeUpdate("DELETE FROM " + prefix + "points "
+                    + "WHERE id NOT IN (SELECT t.min_id FROM ("
+                    + "SELECT MIN(id) AS min_id FROM " + prefix + "points GROUP BY playerID, category) t )");
             statement.executeUpdate("ALTER TABLE " + prefix + "points "
                     + "CHANGE COLUMN playerID profileID CHAR(36) NOT NULL, "
-                    + "MODIFY COLUMN category VARCHAR(255) NOT NULL, " + "DROP PRIMARY KEY," + "DROP COLUMN id "
-                    + "ADD PRIMARY KEY (profileID, category) "
+                    + "MODIFY COLUMN category VARCHAR(255) NOT NULL, "
+                    + "DROP PRIMARY KEY,"
+                    + "DROP COLUMN id, "
+                    + "ADD PRIMARY KEY (profileID, category), "
                     + "ADD FOREIGN KEY (profileID) REFERENCES " + prefix + "profile (profileID) ON DELETE CASCADE");
-            statement.executeUpdate("DELETE FROM " + prefix + "tags WHERE id NOT IN "
-                    + "(SELECT MIN(id) FROM " + prefix + "tags GROUP BY playerID, tag)");
+            statement.executeUpdate("DELETE FROM " + prefix + "tags "
+                    + "WHERE id NOT IN (SELECT t.min_id FROM ("
+                    + "SELECT MIN(id) AS min_id FROM " + prefix + "tags GROUP BY playerID, tag) t )");
             statement.executeUpdate("ALTER TABLE " + prefix + "tags "
                     + "CHANGE COLUMN playerID profileID CHAR(36) NOT NULL, "
-                    + "MODIFY COLUMN tag VARCHAR(510) NOT NULL, " + "DROP PRIMARY KEY," + "DROP COLUMN id "
-                    + "ADD PRIMARY KEY (profileID, tag) "
+                    + "MODIFY COLUMN tag VARCHAR(510) NOT NULL, "
+                    + "DROP PRIMARY KEY,"
+                    + "DROP COLUMN id, "
+                    + "ADD PRIMARY KEY (profileID, tag), "
                     + "ADD FOREIGN KEY (profileID) REFERENCES " + prefix + "profile (profileID) ON DELETE CASCADE");
-            statement.executeUpdate("DELETE FROM " + prefix + "player WHERE id NOT IN "
-                    + "(SELECT MIN(id) FROM " + prefix + "player GROUP BY playerID)");
+            statement.executeUpdate("DELETE FROM " + prefix + "player "
+                    + "WHERE id NOT IN "
+                    + "(SELECT t.min_id FROM (SELECT MIN(id) AS min_id FROM " + prefix + "player GROUP BY playerID) t)");
             statement.executeUpdate("ALTER TABLE " + prefix + "player "
-                    + "MODIFY COLUMN playerID CHAR(36) NOT NULL, " + "MODIFY COLUMN conversation VARCHAR(510), "
+                    + "MODIFY COLUMN playerID CHAR(36) NOT NULL, "
+                    + "MODIFY COLUMN conversation VARCHAR(510), "
                     + "ADD COLUMN active_profile CHAR(36) NOT NULL DEFAULT playerID AFTER playerID, "
```

Comment:
This line is causing an error too `java.sql.SQLSyntaxErrorException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'playe
rID AFTER playerID, ADD FOREIGN KEY (active_profile) REFERENCES bq_profile ' at line 1`

Reply:
This does not make sense to me, as there is no splited `playe rID`